game/scripts/vscripts/heroes/tusk/frozen_sigil.lua

```diff
@@ -0,0 +1,98 @@
+---@class tusk_frozen_sigil_lua : CDOTA_Ability_Lua
+tusk_frozen_sigil_lua = class({})
+---@override
+function tusk_frozen_sigil_lua:OnSpellStart()
+  local caster = self:GetCaster()
+  local sigil_duration = self:GetSpecialValueFor("sigil_duration")
+  local unit = CreateUnitByName("npc_dota_tusk_frozen_sigil"..self:GetLevel(),caster:GetAbsOrigin(),false,caster,caster:GetPlayerOwner(),caster:GetTeamNumber())
+  unit:SetControllableByPlayer(caster:GetPlayerOwnerID(),false)
+  --Transfer this ability to the sigil, so there can't be nil references when this is stolen
+  local ability = unit:AddAbility(self:GetAbilityName())
+  ability:SetLevel(self:GetLevel())
+  -- Default modifier, can't be recreated.
+  unit:AddNewModifier(caster,ability,"modifier_kill",{duration = sigil_duration})
+  unit:AddNewModifier(caster,ability,"modifier_tusk_sigil_slow_aura",{duration = sigil_duration})
+end
+
+LinkLuaModifier("modifier_tusk_sigil_slow_aura","heroes/tusk/frozen_sigil.lua",LUA_MODIFIER_MOTION_NONE)
+---@class modifier_tusk_sigil_slow_aura : CDOTA_Modifier_Lua
+modifier_tusk_sigil_slow_aura = class({})
+
+function modifier_tusk_sigil_slow_aura:OnCreated()
+  if IsServer() then
+    self:GetParent():EmitSound("Hero_Tusk.FrozenSigil")
+    local particle = ParticleManager:CreateParticle("particles/units/heroes/hero_tusk/tusk_frozen_sigil.vpcf", PATTACH_ABSORIGIN_FOLLOW, self:GetParent())
+  end
+end
+-- This also makes it permanent
+---@override
+function modifier_tusk_sigil_slow_aura:IsAura() return true end
+---@override
+function modifier_tusk_sigil_slow_aura:IsDebuff() return false end
+---@override
+function modifier_tusk_sigil_slow_aura:GetAuraSearchTeam()
+  return DOTA_UNIT_TARGET_TEAM_ENEMY
+end
+---@override
+function modifier_tusk_sigil_slow_aura:GetAuraRadius()
+
+  return self:GetAbility():GetSpecialValueFor("sigil_radius")
```

Comment:
I dont think an existing sigil changes if the ability is leveled, if this is the case I would move this to OnCreated.